reference-app/app.js

```diff
@@ -285,6 +285,9 @@ mainModule.controller('MainCtrl', ['$scope', '$http', '$q', 'remote', 'config',
     $scope.setTemperatureValue = function (device, property, value) {
         setDeviceProperty(device, property, { temperature: value, units: property.units }).then(info => {
             property.temperature = info.temperature;
+            if (!!property.units) {
```

Comment:
Did you mean if (!!info.units)? or perhaps: if (!property.units), with a single '!'?
Otherwise you are clobbering a non-null, non-undefined value with another value.

Reply:
Aha, I did mean !!info.units.